src/rtklib.h

```diff
@@ -542,6 +543,8 @@ typedef struct {        /* observation data record */
     double L[NFREQ+NEXOBS]; /* observation data carrier-phase (cycle) */
     double P[NFREQ+NEXOBS]; /* observation data pseudorange (m) */
     float  D[NFREQ+NEXOBS]; /* observation data doppler frequency (Hz) */
+    /* only used if refpos == POSOPT_RINEX_DYN */
+    double refpos[3];   /* station coordinates associated with this observation */
```

Comment:
nit, not sure if this is a good place to store `refpos`. This struct is for observations, maybe we should keep it as it is and create a struct with `refpos` and `gtime_t` as members, and then use it where it is needed?

Reply:
Trust me, I spent a long time trying to find a better place for it.  The issue is that implementing a "nicer" solution requires  modifications to multiple function prototypes in rtklib.h which could potentially break compatibility with other people who have made their own custom modifications to RTKLIB.  If you can come up with a better approach then I would like to see it!

Reply:
yeah, it is a bit tricky. I am thinking of adding an array of a structure with `refpos` and `gtime_t` in the `obs_t` struct, and then query the `refpos` when a time is given before https://github.com/swift-nav/RTKLIB/blob/77308a72dc6a011c2a45a0f929707f389bf9a830/src/postpos.c#L396, and then update the `rtk->opt.rb` before `rtkpos` is called.
Don't know whether that's a better solution, feel free to use the current implementation.

Reply:
I tried something like that at one point.  The issue is that callers won't know that they need to free the additional array.